Document HomematicIP button events (#45416)
home-assistant/home-assistant.github.io: source/_integrations/homematicip_cloud.markdown

      accesspoint_id: 3014xxxxxxxxxxxxxxxxxxxx
```

## Button events

> Devices with physical buttons expose an event entity per button channel. You can use these to trigger automations on a short press, a long press start, or a long press stop.

Reviewer: ```suggestion
Devices with physical buttons expose an event entity per button channel. You can use these to trigger automations on a short release, a long press, or a long press release.
```
are "short press, a long press start, or a long press stop" what is called "short release, a long press, or a long press release" in the UI? I did not find these start and stop in the strings.

{% important %}

> If a button is directly paired to an actuator inside the Homematic IP app (called *Direktverknüpfung* in the German app), the cloud doesn't see the press, so Home Assistant can't react to it either. To use the button in Home Assistant, either remove the direct pairing in the Homematic IP app, or add an automation in the Homematic IP app that references the button. The cloud then forwards the press to Home Assistant.

Reviewer: ```suggestion
If a button is directly paired to an actuator inside the Homematic IP app (*Direct Device Connection*, called *Direktverknüpfung* in the German app), the cloud doesn't see the press, so Home Assistant can't react to it either. To use the button in Home Assistant, either remove the direct pairing in the Homematic IP app, or add an automation in the Homematic IP app that references the button. The cloud then forwards the press to Home Assistant.
```
Is this the same as "Direct Device Connection"?

{% endimportant %}

{% note %}
The cloud doesn't deliver a dedicated double-press event. A double-press arrives as two consecutive `short_press` events, so double-press automations are built on top of the same event entity, as shown below.
{% endnote %}

### Trigger an action on a short press

> ```yaml

Reviewer: ```suggestion
{% example %}
```

automation:
  - alias: "Toggle living room light on button 3 short press"
    triggers:
      - trigger: state
        entity_id: event.wandtaster_6_fach_button_3
        attribute: event_type
        to: short_press
    actions:
      - action: light.toggle
        target:
          entity_id: light.living_room

> ```

Reviewer: ```suggestion
{% endexample %}
```
We now have an example liquid tag we can use instead of the yaml tag.
https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L175

### Trigger an action on a double press

The cloud delivers a double press as two consecutive `short_press` events. Use `wait_for_trigger` with a short timeout to detect the second press:

> ```yaml

Reviewer: ```suggestion
{% example %}
```

automation:
  - alias: "Activate movie scene on button 3 double press"
    triggers:
      - trigger: state
        entity_id: event.wandtaster_6_fach_button_3
        attribute: event_type

>         to: short_press

Reviewer: ```suggestion
        to: short_release
```
Are these YAML keys still correct? In core, I see long_press, long_release, short_release.

    actions:
      - wait_for_trigger:
          - trigger: state
            entity_id: event.wandtaster_6_fach_button_3
            attribute: event_type
            to: short_press
        timeout: "00:00:00.500"
        continue_on_timeout: false
      - action: scene.turn_on
        target:
          entity_id: scene.movie_night

> ```

Reviewer: ```suggestion
{% endexample %}
```

### Dim a light while holding a button

The button reports `long_press_start` when held and `long_press_stop` when released. Use a `repeat` loop that keeps dimming until the release event fires:

> ```yaml

Reviewer: ```suggestion
{% example %}
```

automation:
  - alias: "Dim living room while holding button 3"
    triggers:
      - trigger: state
        entity_id: event.wandtaster_6_fach_button_3
        attribute: event_type
        to: long_press_start
    actions:
      - repeat:
          sequence:
            - action: light.turn_on
              target:
                entity_id: light.living_room
              data:
                brightness_step_pct: -5
            - delay: "00:00:00.200"
          while:
            - condition: state
              entity_id: event.wandtaster_6_fach_button_3
              attribute: event_type
              state: long_press_start

> ```

Reviewer: ```suggestion
{% endexample %}
```
